src/react-imgix.jsx

```diff
@@ -220,14 +220,20 @@ class ReactImgix extends Component {
       defaultAttributeMap,
       this.props.attributeConfig
     );
-    const childProps = Object.assign({}, this.props.htmlAttributes, {
-      [attributeConfig.sizes]: this.props.sizes,
-      className: this.props.className,
-      width: width <= 1 ? null : width,
-      height: height <= 1 ? null : height,
-      [attributeConfig.src]: src,
-      ref: el => (this.imgRef = el)
-    });
+    const childProps = Object.assign(
+      {},
+      {
+        [attributeConfig.sizes]: this.props.sizes,
+        className: this.props.className,
+        width: width <= 1 ? null : width,
+        height: height <= 1 ? null : height,
+        [attributeConfig.src]: src,
+        ref: el => {
+          this.imgRef = el;
+        }
+      },
```

Comment:
The downside to this is that users could potentially overwrite any of the other properties included in this object blob via `htmlAttributes` if not careful. To prevent this, we could keep the original order but pull out the `ref` assignment before `htmlAttributes`. Thoughts?

Reply:
For problems like these I like to go with the second option you suggested.